app/controllers/providers/home_address/home_address_selections_controller.rb

```diff
@@ -0,0 +1,59 @@
+module Providers
+  module HomeAddress
+    class HomeAddressSelectionsController < ProviderBaseController
+      include AddressHandling
+
+      def show
+        return redirect_to back_path unless address.postcode
+
+        @addresses = address_lookup.result
+        titleize_addresses
+        filter_home_addresses if applicant.home_address.building_number_name.present?
+        @address_collection = collect_addresses
+        @form = Addresses::AddressSelectionForm.new(model: address)
+      end
+
+      def update
+        if params[:address_selection][:list]
+          @addresses = build_addresses_from_form_data
+          @address_collection = collect_addresses
+          @form = Addresses::AddressSelectionForm.new(address_selection_form_params)
+        else
+          @form = Addresses::AddressForm.new(address_form_params)
+        end
+
+        render :show unless save_continue_or_draft(@form)
+      end
+
+    private
+
+      def address
+        applicant.home_address || applicant.build_address
```

Comment:
Sorry for the late comment, but shouldn't the `build_address` have a `location: "home"` appended?

Reply:
Ah, nope, I see you add it at the `address_selection_form_params` stage - ignore this! 😄